RFCs/0012-Test-Adapter-Lookup.md

```diff
@@ -0,0 +1,91 @@
+| Extensions directory     | Adapters packaged with VS, available in the Extensions directory               |
+| VSIX                     | Adapter ships as a vsix.                                                       |
+
+## Proposed Changes
+Here are the proposed changes based on the sources available for lookup of the adapters.
+
+### Priority order for Lookup
+Following is the priority order for the lookup of the adapters
+
+| Priority | Source                   |
+|----------|--------------------------|
+| 1        | Nuget                    |
+| 2        | TestAdapterPath          |
+| 3        | Project output directory |
+| 4        | Extensions directory     |
+| 5        | VSIX                     |
+
+### Details for lookup
+
+#### Nuget
+This lookup is based on the nuget package references. This works only for IDE and not for CLI, as this information is only available in IDE.
+IDE invokes all the test adapters with all the test projects in the solution.
+If multiple versions of same adapter, or multiple adapters with same URI found, TestPlatform will throw a warning.
+
+* Recommendation : Use one version of the given adapter in the solution.
+
+* Best Practice : Define a constant for nuget package version of an adapter in one file and use it across all the projects.
+This also helps make the upgrade experience to a newer version of adapter seamless.
+For example, open source projects like
+[Roslyn](https://github.com/dotnet/roslyn) and
+[VSTest](https://github.com/microsoft/vstest) use *props file to define the version of the nuget packages.
+
+#### TestAdapterpath
+TestAdapterPath can be given via runsettings or as an argument in case of cli.
+All the test sources will be run against all the test adapters found in the test adapter path.
+If multiple versions of same adapter, or multiple adapters with same URI found, TestPlatform will throw a warning.
+
+* Recommendation : Have one version of an adapter in the test adapter path.
+
+#### Project Output Directory
+Some adapters packaged as nuget get copied to the output directory of the project.
+
+Since IDE picks up the adapters from the nuget package location, there is no need to look for adapters in the project output directory.
+Hence, this project output directory will be probed only in case of cli. Further, cli will probe only when testAdapterPath is not specified.
+
+* Recommendation : Use /testadapterpath option with cli. With multiple copies of adapters in the project output directories, each project needs to be run in isolation, which may impact test execution performance via cli.
+
+##### Open for discussion
+Should the adapters even be copied to the project output directory ?
+
+#### Extensions directory
+All the adapters in the extensions directory get loaded by default and all the test sources based on the file extensions are passed to these adapters. Test runner will provide the diagnostics information about adapters used for the test projects.
```

Comment:
might be worth calling out the assembly pattern change since that is different from TP V1 for adapters in extensions folder. Or does that not apply here?